public/javascripts/SidewalkGallery/src/validation/ValidationMenu.js

```diff
@@ -0,0 +1,128 @@
+/**
+ * A Validation Menu to be appended to a Card for validation purposes.
+ * 
+ * @param uiCardImage The html element to append the validation menu to.
+ * @param cardProperties Properties of the label the validation menu is being appended to
+ * @returns {ValidationMenu}
+ * @constructor
+ */
+function ValidationMenu(uiCardImage, cardProperties) {
+    let resultOptions = {
+        "Agree": 1, 
+        "Disagree": 2,
+        "NotSure": 3
+    };
+
+    let currSelected = null;
+
+    const overlayHTML = `
+        <div id="gallery-validation-button-holder">
+            <button id="gallery-card-agree-button" class="validation-button">Agree</button>
+            <button id="gallery-card-disagree-button" class="validation-button">Disagree</button>
+            <button id="gallery-card-not-sure-button" class="validation-button">Not Sure</button>
+        </div>
+    `;
+
+    let overlay = $(overlayHTML);
+
+    let agreeButton = overlay.find("#gallery-card-agree-button");
+    let disagreeButton = overlay.find("#gallery-card-disagree-button");
+    let notSureButton = overlay.find("#gallery-card-not-sure-button");
+
+    function _init() {
+        // TODO: compress this code
+        agreeButton.click(function() {
+            if (currSelected) {
+                currSelected.attr('class', 'validation-button');
+            }
+
+            currSelected = agreeButton;
+            agreeButton.attr('class', 'validation-button-selected');
+
+            validateLabel("Agree");
+        });
+        
+        disagreeButton.click(function() {
+            if (currSelected) {
+                currSelected.attr('class', 'validation-button');
+            }
+
+            currSelected = disagreeButton;
+            disagreeButton.attr('class', 'validation-button-selected');
+
+            validateLabel("Disagree");
+        });
+        
+        notSureButton.click(function() {
+            if (currSelected) {
+                currSelected.attr('class', 'validation-button');
+            }
+
+            currSelected = notSureButton;
+            notSureButton.attr('class', 'validation-button-selected');
+
+            validateLabel("NotSure");
+        });
+
+        uiCardImage.appendChild(overlay[0]);
+    }
+
+    /**
+     * Consolidate data on the validation and submit as a POST request.
+     * 
+     * @param action Validation result.
+     * @private
+     */
+    function validateLabel(action) {
+        console.log("validate method called");
+
+        // TODO: do we need this log?
```

Comment:
I don't know. I was thinking it was redundant because we already log it into the validation table. But maybe for the purpose of logging the click interaction itself it makes sense?

Reply:
Oh yeah we should definitely log the click interaction.